llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -288,6 +290,26 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineVectorInserts(
   return true;
 }
 
+bool AIE2PreLegalizerCombinerImpl::tryToCombineVExtractElt(
+    MachineInstr &MI) const {
```

Comment:
Please add an assert to check the specific "opcode" we expected for the MI here. 
What I understand is this works for specific base opcode. This will improve readability for future and right away tell which Opcode this function should be called for. 

Similar this is done in multiple target in legalizer or instruction select.

Reply:
Like other combines in this file, we are calling this function for multiple intrinsic IDs. However, I don't see the necessity for this, as there are no assertions in the other combines either.
```
case Intrinsic::aie2_vextract_elem8_I512:
case Intrinsic::aie2_vextract_elem16_I512:
case Intrinsic::aie2_vextract_elem32_I512: {
  return tryToCombineVExtractElt(MI);
}